source/shared/core_results.cpp

```diff
@@ -1346,8 +1346,8 @@ SQLRETURN sqlsrv_buffered_result_set::to_same_string( SQLSMALLINT field_index, _
     return r;
 }
 
-SQLRETURN sqlsrv_buffered_result_set::wide_to_system_string( SQLSMALLINT field_index, _Out_ void* buffer, SQLLEN buffer_length, 
-                                                             _Out_ SQLLEN* out_buffer_length )
+SQLRETURN sqlsrv_buffered_result_set::wide_to_system_string( _In_ SQLSMALLINT field_index, _Inout_updates_bytes_to_(buffer_length, *out_buffer_length) void* buffer, _In_ SQLLEN buffer_length,
+                                                             _Inout_ SQLLEN* out_buffer_length )
```

Comment:
this one is a little tricky. In line 1432, temp_string is copied to buffer using memcpy_s, but only if (to_copy > 0) is true (line 1431). Later, we need to null terminate buffer. If to_copy == 0, that means memcpy_s never took place, but now we need to read buffer[to_copy] to put in the null_terminator, thus _Inout_ is used